### ISSUE TITLE
Setting the correct gain

### ISSUE THREAD

**Issue by milanico2309** [TheDIYGuy999/MRSC_Adapter_WLtoys_5Pin_Servo#1]
Hi,

at first let me thank you for sharing this project. i succefully implemented it in my a959 setup. My last problem is, that i can't find the correct gain setting. i tried your default 20, but also 5, 1, 0.1 and 0.01. The Systems is still too nervous so my car does wobbly lines when i wan't to go straight. It's like the Camaro in your other video. (https://www.youtube.com/watch?v=IPve7QpdLBc)

What can i do to fix this?

M

**Comment by TheDIYGuy999** [comment]
You're welcome. The wobble may be caused by too much backlash in your steering mechanism. Are the resistor and capacitor values identical with my setup? Otherwise the reaction speed of the output signal may be wrong and cause wobble.

**Comment by milanico2309** [comment]
Yes, i checked the specs. of the filter-components again. The values are as specified. I fiddelt around with the code and noticed something. If i do ` int turnRateMeasured = (yaw_rate * 127) * 0.3;` istead of ` int turnRateMeasured = (yaw_rate * 127) * mrscGain/100;` with a gain of 30, it works. so it may be a datatype problem screwing up the calculation. (because turnRateMeasured is an integer, so the diversion alway returns full numbers e.g. 1 or 0). But i may be wrong, had no chance to debug yet...

**Comment by TheDIYGuy999** [comment]
Strange, but indeed, an overflow could occur, depending on your gain.
try: long turnRateMeasured = (yaw_rate * 127) * mrscGain / 100;
My current gain is 15 BTW.

**Comment by owerlordalex** [comment]
![1586338321628](https://user-images.githubusercontent.com/63281949/78768729-dafa5a80-797b-11ea-8c0a-6194a9a70e65.jpg)
It's working great! Thank you
